### PR TITLE
fix(prometheus): create legacy endpoints instead of checking if they exist

### PR DESCRIPTION
**Pull request by Duologic**
std.objectHas didn't find the hidden elements, causing the configmaps to be empty and removed. this should restore that.

### CODE REVIEW

File: prometheus/prometheus.libsonnet [hunk 1]
```diff
@@ -41,7 +44,7 @@ local kausal = import 'ksonnet-util/kausal.libsonnet';
       }),
     ]
     + (
-      if std.objectHas(this, 'prometheusAlerts') && std.prune(this.prometheusAlerts) != {}
+      if std.prune(this.prometheusAlerts) != {}
```
Comment (julienduchesne) [inline prometheus/prometheus.libsonnet:47]: Shouldn't you prune while defining your local above (l38)? You prune here while checking for an empty definition but you don't while creating your yaml. Isn't that inconsistent? How often do these maps have empty members?

**Review by julienduchesne — APPROVED** [review]

**Review by mplzik — APPROVED** [review]
LGTM

File: prometheus/prometheus.libsonnet (hunk 1 above)
Comment (Duologic) [reply prometheus/prometheus.libsonnet:47]: No, empty members sometimes have meaning, I have fallen in this trap before. :grimacing:

Comment (julienduchesne) [reply prometheus/prometheus.libsonnet:47]: Why prune at all then?

**Pull request merged**